File: examples/java/kora-java-scheduling-jdk/src/main/java/ru/tinkoff/kora/example/scheduling/jdk/ConfigScheduler.java
package ru.tinkoff.kora.example.scheduling.jdk;

import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleAtFixedRate;

@Component
public final class ConfigScheduler {

    private int state = 0;

    @ScheduleAtFixedRate(config = "scheduling.jobs.fix-rate")
    void schedule() {
        state++;
    }

    public int getState() {
        return state;
    }
}

File: examples/kotlin/kora-kotlin-scheduling-jdk/src/main/kotlin/ru/tinkoff/kora/kotlin/example/scheduling/jdk/ConfigScheduler.kt
package ru.tinkoff.kora.kotlin.example.scheduling.jdk

import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleAtFixedRate
import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleOnce
import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleWithFixedDelay
import java.time.temporal.ChronoUnit

@Component
class ConfigScheduler {
    var state = 0
        private set

    @ScheduleAtFixedRate(config = "scheduling.jobs.fix-rate")
    fun schedule() {
        state++
    }
}

